lib/prefab/feature_flag_client.rb

```diff
@@ -19,27 +19,48 @@ def feature_is_on?(feature_name)
     def feature_is_on_for?(feature_name, lookup_key, attributes: {})
       @base_client.stats.increment("prefab.featureflag.on", tags: ["feature:#{feature_name}"])
 
-      return is_on?(get(feature_name, lookup_key, attributes))
+      return is_on?(_get(feature_name, lookup_key, attributes, default: false))
     end
 
-    def get(feature_name, lookup_key=nil, attributes={})
-      feature_obj = @base_client.config_client.get(feature_name)
-      config_obj = @base_client.config_client.get_config_obj(feature_name)
-      return nil if feature_obj.nil? || config_obj.nil?
-      variants = config_obj.variants
-      evaluate(feature_name, lookup_key, attributes, feature_obj, variants)
-    end
+    def get(feature_name, lookup_key=nil, attributes={}, default: false)
+      variant = _get(feature_name, lookup_key, attributes, default: default)
 
-    def evaluate(feature_name, lookup_key, attributes, feature_obj, variants)
-      value_of_variant(get_variant(feature_name, lookup_key, attributes, feature_obj, variants))
+      value_of_variant_or_nil(variant, default)
     end
 
     private
 
+    def value_of_variant_or_nil(variant_maybe, default)
+      if variant_maybe.nil?
+        default != Prefab::Client::NO_DEFAULT_PROVIDED ? default : nil
+      else
+        value_of_variant(variant_maybe)
+      end
+    end
+
+    def _get(feature_name, lookup_key=nil, attributes={}, default:)
+      feature_obj = @base_client.config_client.get(feature_name, default)
+      config_obj = @base_client.config_client.get_config_obj(feature_name)
+
+      return nil if feature_obj.nil? || config_obj.nil?
+
+      if feature_obj == !!feature_obj
+        return feature_obj
+      end
```

Comment:
NOTE: we have a couple `var == !!var` evaluations. These are to support
pseudo-feature flags which are actually just a config with a value of
`true` or `false`.

We can probably drop support for this, but it is currently explicitly
mentioned in the documentation so it might be worth supporting.